src/core/server/config/integration_tests/config_deprecation.test.ts

```diff
@@ -23,13 +23,17 @@ describe('configuration deprecations', () => {
     }
   });
 
-  it('should not log deprecation warnings for default configuration', async () => {
+  it('should not log deprecation warnings for default configuration that is not one of `logging.verbose`, `logging.quiet` or `logging.silent`', async () => {
```

Comment:
`CliArgs` interface includes `quiet: boolean` and `silent: boolean`. I wasn't sure of the best way to handle that or if the new behavior of deprecating those configuration options is what we want.

I'm open to suggestions on how to proceed both with the behavior of the cli args and to refactoring/modifying the test.

Reply:
our `CliArgs` is mostly a shortcut for configuration values defined in `kibana.yml` file. Since we are deprecating `logging.verbose/quiet/silent`, we should deprecate their counterparts from CLI, I suppose. Otherwise, there might be a conflict between CLI values and `kibana.yml` config values (For example, a user can specify an appender for the `http` context but run Kibana via CLI with `silent` flag).

Reply:
I think that asking people who use the `--verbose` flag to switch to `--logging.root.level=debug --logging.root.appenders[0]=default --logging.root.appenders[1]=console`, and then also asking them to keep track of when the legacy logging is removed and stop using the default appender is an unnecessarily bad experience when we could keep the `--verbose` flag as a shortcut for these options.

Reply:
Same with the other logging level flags

Reply:
Response from the ops team:
"The logging related flags have always been focused on the CLI logging, which feels like it could be persisted as a convenience going forward"

Our current recommendation is to replace `verbose` with `--logging.root.level=debug --logging.root.appenders[0]=default --logging.root.appenders[1]=console` but we're removing the default appender when we remove the legacy logging system.

We'll have to update our recommended replacement with:
`--logging.root.level=debug --logging.root.appenders[1]=console`, which is doable I guess.

Reply:
My request from https://github.com/elastic/kibana/pull/94238#discussion_r591754861 is that we keep this convenience feature for folks who are trying to get more info out of Kibana with the `--verbose` flag rather than expecting that they read the docs to learn the new incantation to replicate the `--verbose` behavior.

Reply:
> My request from #94238 (comment) is that we keep this convenience 

I haven't checked the implementation, but I'd expect the `verbose` flag continues working in `7.x`, but the Core logs a deprecation warning. Starting from 8.0 `verbose` is deleted, users use `--logging.root.level=debug` instead. Do you see any problems with this approach?

Reply:
Yeah, this is really just a UX thing, and I think if it's easy enough to support `--verbose` as an alias for `--logging.root.level=debug` then I'd definitely prefer that we keep it.

Reply:
If we _did_ unfold the flags, would we want to set the appender to `console` by default? What if someone wanted to send the logs to file?

We'd need to then add additional support for a `file` appender and that opens up a whole new can of worms.

We don't need to necessarily add support for writing to file from the start, we could do that later as an enhancement.

Reply:
> If we _did_ unfold the flags, would we want to set the appender to `console` by default? What if someone wanted to send the logs to file?

I don't think that `--verbose` should do anything but change the level

> We'd need to then add additional support for a `file` appender

Can we just override the level of all the appenders when you use one of these flags from the CLI? Feels like that will mostly just apply to the console appender and if you're configuring a file appender in kibana.yml and want different levels for each appender then you've already read the docs on how that works and don't really need the shortcut flag.

Reply:
> Feels like that will mostly just apply to the console appender...
Agreed, I wouldn't want to assume too much and override _every_ custom appender's log level since, as you point out, if there _are_ custom appenders, they're there for a reason.

Side note: What should we do about the `--json` flag? That's also going to log a deprecation warning.

Reply:
Do we ever document these CLI flags? Probably it's fine to ditch some of them (like logFile)?

Reply:
@spalger how would you feel about us renaming these cli args to mimic the levels we're using in the new logging system? For example: `verbose` -> `logAll` (maps to KP logging level of `all` etc.

Reply:
> Do we ever document these CLI flags?

Only in the [logging README](https://github.com/elastic/kibana/blob/master/src/core/server/logging/README.mdx#logging-configuration-via-cli).

Adding [user-facing docs](https://github.com/elastic/kibana/issues/57502) is my next priority once this PR is done.

Reply:
This is definitely a complex case to wrap your head around. I think I'm understanding it more now and I think we should optimize for:
- The least number of disruptive surprises for the user
- A convenient experience for debugging problems
- Removing unused features

Maybe we can start on what we agree with, I think this seems to line up with what we're thinking:
- Keep support for `--verbose` and `--silent` and deprecate all other convenience logging CLI flags
- For the flags we're maintaining, we can unfold the flags to write both the appropriate configs for legacy and new logging. For example, setting `--verbose` would set both `logging.level` (legacy) and `logging.root.level` (new). We shouldn't change the root appender at all in this case, since just specifying `--verbose` should not opt you in for new logging appenders.
- For the options we're deprecating, we should always log a deprecation warning if they are specified, even if they're specified the same value as the default values. This may require some changes to the legacy logging config schema and in `serve.js` to not set the defaults here so that we can differentiate between `--logging.silent=false` and not setting this flag at all: https://github.com/elastic/kibana/blob/8fc5d8ba30d242cdc9d3febb8e75cbc8d83af970/src/cli/serve/serve.js#L115-L118
- `--verbose` and `--silent` should take precedence over configuration yamls

Any other parts that I'm missing?

Reply:
> Do we ever document these CLI flags?

@restrry We also document them in the `./bin/kibana --help`

> @spalger how would you feel about us renaming these cli args to mimic the levels we're using in the new logging system?

@TinaHeiligers I'm not staunchly opposed to new flags, but I think that `--verbose` is a pretty universal flag that just tells a CLI tool to "give me a much info as possible" so I'd like to maintain support for that if possible

> Any other parts that I'm missing?

That looks good to me @joshdover